build_web_compilers/lib/src/dev_compiler_stack_trace/stack_trace_mapper.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Standalone utility that manages loading source maps for all Dart scripts
+/// on the page compiled with DDC.
+///
+/// Example JavaScript usage:
+/// $dartStackTraceUtility.addLoadedListener(function() {
+///   // All Dart source maps are now loaded. It is now safe to start your
+///   // Dart application compiled with DDC.
+///   dart_library.start('your_dart_application');
+/// })
+///
+/// If $dartStackTraceUtility is set, the dart:core StackTrace class calls
+/// $dartStackTraceUtility.mapper(someJSStackTrace)
+/// to apply source maps.
+///
+/// This utility can be compiled to JavaScript using Dart2JS while the rest
+/// of the application is compiled with DDC or could be compiled with DDC.
+
+@JS()
+library stack_trace_mapper;
+
+import 'package:js/js.dart';
+import 'package:path/path.dart' as path;
+import 'package:source_maps/source_maps.dart';
+import 'package:source_span/source_span.dart';
+import 'package:stack_trace/stack_trace.dart';
+
+import 'package:build_web_compilers/src/dev_compiler_stack_trace/source_map_stack_trace.dart';
+
+typedef void ReadyCallback();
+
+/// Global object DDC uses to see if a stack trace utility has been registered.
+@JS(r'$dartStackTraceUtility')
+external set dartStackTraceUtility(DartStackTraceUtility value);
+
+@JS(r'$dartLoader.rootDirectories')
+external List get rootDirectories;
+
+typedef String StackTraceMapper(String stackTrace);
+typedef dynamic SourceMapProvider(String modulePath);
+typedef void SetSourceMapProvider(SourceMapProvider provider);
+
+@JS()
+@anonymous
+class DartStackTraceUtility {
+  external factory DartStackTraceUtility(
+      {StackTraceMapper mapper, SetSourceMapProvider setSourceMapProvider});
+}
+
+@JS('JSON.stringify')
+external String _stringify(dynamic json);
+
+/// Source mapping that is waits to parse source maps until they match the uri
```

Comment:
This sentence doesn't read correctly to me...

Reply:
This is just copied from the sdk, but I fixed up the wording (removed the `is`)